provider/src/lib.rs

```diff
@@ -1,7 +1,58 @@
-//! Data providers for time zone data
+//! Providers for time zone data
 //!
-//! This crate aims to provide a variety of data providers
-//! for time zone data.
+//! Let's talk about time zone data everyone!
+//!
+//! At a high level, the `timezone_provider` crate provides a set of traits along with a few
+//! implementations of those traits. The general intention here is to make providing time zone
+//! data as agnostic and easy as possible.
+//!
+//! This crate is fairly "low level" at least as far as date and time needs are concerned. So
+//! we'll cover the basic overview of the trait and some of the general implementations of
+//! those traits, and then we will go on a bit further of a dive for the power users that
+//! are interested in implementing their own provider or is just really curious about what
+//! is going on.
+//!
+//! ## Available providers
+//!
+//! Below are a list of currently avaiable time zone providers.
+//!
+//! - `ZoneInfo64TzdbProvider` (enable with `zoneinfo64` features flag)
```

Comment:
suggestion: probably include a one sentence description of these.

Include the fact that CompiledTzdbProvider still has a runtime cost of parsing tzif files, but it caches.

Also probably ensure the docs for those providers include that info too.